tracer/README.MD

```diff
@@ -99,6 +99,12 @@ You can use Rider and CLion to develop on macOS. Building and testing can be don
 .\build.sh BuildAndRunIntegrationTests
 ```
 
+## Additional Technical Documentation
+
+* [Implementing an automatic instrumentation](./src/Datadog.Trace/ClrProfiler/README.md)
```

Comment:
Maybe as a separate PR so you don't invoke the CI gods, but can we move the ClrProfiler/README.md and DuckTyping/README.md to the `/docs` folder? I think that would aid in discoverability

Reply:
I guess this would be a team-wide convention discussion. My preference would be to have most documentation living _only_ in the docs folder versus spread out in the projects they affect

Reply:
Done in #2317